File: src/main/java/cf/dogo/utils/WebUtils.java
package cf.dogo.utils;

import java.net.URLEncoder;
import java.util.HashMap;
import java.util.Map;

public class WebUtils {
    public static String get(String url, HashMap<String, Object> args){
        if(!url.endsWith("/")) url+="/";
        args.keySet().forEach(k -> {
            try {
                args.replace(k, URLEncoder.encode(args.get(k).toString(), "UTF-8"));
            }catch (Exception ex){}
        });
        if(!args.isEmpty()){
            url+="?";
            for(Map.Entry<String, Object> arg : args.entrySet()){
                url+=arg.getKey()+"="+arg.getValue()+"&";
            }
            url = url.substring(0, url.length()-1);
        }
        return url;
    }
}

File: src/main/kotlin/cf/dogo/utils/WebUtils.kt
package cf.dogo.utils

import java.net.URLEncoder

class WebUtils {
    companion object {
        operator fun get(url: String, vararg args: Pair<String, Any>): String {
            var url = url
            if (!url.endsWith("/")) url += "/"
            if(args.isNotEmpty()){
                url += args
                        .map {
                            Pair(URLEncoder.encode(it.first, "UTF-8"), URLEncoder.encode(it.second.toString(), "UTF-8"))
                        }
                        .joinToString(separator = "&", prefix = "&")
            }
            return url
        }
    }
}